.github/actions/setup-maven-cache/action.yaml

```diff
@@ -46,27 +46,46 @@ runs:
         -D aether.syncContext.named.time=180
         -D maven.artifact.threads=32
         EOF
+    - name: Get cache path and keys
+      env:
+        HASH_FILES: ${{ hashFiles('**/pom.xml') }}
+        # it matters for caching as absolute paths on self-hosted and GitHub runners differ
+        # self-hosted: `/runner/` vs gh-hosted: `/home/runner`
+        KEY_PREFIX: ${{ runner.environment }}-${{ runner.os }}-mvn-${{ inputs.maven-cache-key-modifier }}
+      run: |
+        maven_version=$(./mvnw --version | head -n 1 | sed "s/Apache Maven \([[:digit:]]\.[[:digit:]]\.[[:digit:]]\).*/\1/g")
+        # Only cache release artifacts for Maven version 3.9+
+        # GitHub env variables are used instead of outputs, as outputs are not available in post steps
+        # https://github.com/actions/runner/issues/2009
+        if echo "3.9.0\n$maven_version" | sort --version-sort --check=quiet; then
+          # This is the path used by the `enhancedLocalRepository` set up in the 'Configure Maven' step.
+          # `aether.enhancedLocalRepository.remotePrefix` defaults to 'cached'
+          # `aether.enhancedLocalRepository.releasesPrefix` defaults to 'releases'
+          echo "MAVEN_CACHE_PATH=~/.m2/repository/cached/releases/" >> $GITHUB_ENV
+          echo "MAVEN_CACHE_KEY=${KEY_PREFIX}-${HASH_FILES}" >> $GITHUB_ENV
+          echo "MAVEN_CACHE_KEY_PREFIX=${KEY_PREFIX}" >> $GITHUB_ENV
+        else
+          echo "MAVEN_CACHE_PATH=~/.m2/repository/" >> $GITHUB_ENV
+          echo "MAVEN_CACHE_KEY=${KEY_PREFIX}-full-${HASH_FILES}" >> $GITHUB_ENV
+          echo "MAVEN_CACHE_KEY_PREFIX=${KEY_PREFIX}-full" >> $GITHUB_ENV
+        fi
+      shell: sh
```

Comment:
For smoother reading I'd put this `shell` property before `run` so people know how to interpret the script before trying to comprehend it completely

Reply:
Noted 👍 
Will do in subsequent PR.

Reply:
https://github.com/camunda/camunda/pull/23323/files#diff-848e846785397574218a743c39255aa98ee08ec0098eb75bd9680a963119218cR59